lib/ultrajsonenc.c

```diff
@@ -551,9 +551,20 @@ static void Buffer_AppendIndentUnchecked(JSONObjectEncoder *enc, JSINT32 value)
 {
   int i;
   if (enc->indent > 0)
-    while (value-- > 0)
-      for (i = 0; i < enc->indent; i++)
-        Buffer_AppendCharUnchecked(enc, ' ');
+  {
+    if (enc->indent_chars == NULL)
+    {
+        while (value-- > 0)
+          for (i = 0; i < enc->indent; i++)
+            Buffer_AppendCharUnchecked(enc, ' ');
+    }
+    else
+    {
+        while (value-- > 0)
+          for (i = 0; i < enc->indent; i++)
+            Buffer_AppendCharUnchecked(enc, enc->indent_chars[i]);
```

Comment:
For performance, this should probably be replaced with a `memcpy`, though an extra check in debug mode that this doesn't overrun the buffer would be a good idea. Something like this, I think:

```suggestion
#ifdef DEBUG
          if (enc->end - enc->offset < enc->indent)
          {
            fprintf(stderr, "Ran out of buffer space during Buffer_AppendIndentUnchecked()\n");
            abort();
          }
#endif
          memcpy(enc->offset, enc->indent_chars, enc->indent)
```

(Come to think of it, such checks would be a good idea on the other `memcpy`s as well, but that's for another PR.)

A test for this that's basically a merger between `test_dump_huge_indent` and `test_dump_long_string` would be good as well. Similar variation, arranged such that it should hit a memory boundary. It's too late right now for me to think this through fully though.

Reply:
I forgot that of course `enc->offset` needs to be incremented by `enc->indent` after the copy as well. Fixed that now.

Reply:
#529 adds a `Buffer_memcpy` function that could be used instead once merged.